android/src/play/java/com/dooboolab/RNIap/RNIapModule.kt

```diff
@@ -180,73 +181,72 @@ class RNIapModule(
         ) {
             val skuList = ArrayList<String>()
             for (i in 0 until skuArr.size()) {
-                val sku = skuArr.getString(i)
-                if (sku is String) {
+                if (skuArr.getType(i) == ReadableType.String) {
+                    val sku = skuArr.getString(i)
                     skuList.add(sku)
                 }
             }
             val params = SkuDetailsParams.newBuilder()
             params.setSkusList(skuList).setType(type)
-            billingClient.querySkuDetailsAsync(
+            billingClient?.querySkuDetailsAsync(
```

Comment:
Should we reject if `billingClient` is null?

Reply:
I fixed it so it never gets into that case